src/components/JobsLogs.vue

```diff
@@ -0,0 +1,397 @@
+        return "Success";
+      case 4:
+        return "Error";
+      case 5:
+        return "Cancelled";
+      case 824:
+        return "Cancelling";
+      default:
+        return "Unknown";
+    }
+  }
+
+  formatDateString(dateString: string): string {
+    return formatDateString(dateString);
+  }
+
+  getFirstArg(job: IJob): string {
+    return job.args && job.args.length > 0 ? job.args[0] : "";
+  }
+
+  getEndTime(job: any): string {
+    // Find the timestamp with status 3 (success), 4 (error), or 5 (cancelled)
+    const endTimestamp = job.timestamps?.find((ts: any) =>
+      [3, 4, 5].includes(ts.status),
+    );
+
+    if (endTimestamp) {
+      return formatDateString(endTimestamp.time);
+    }
+
+    return job.status === 2 ? "Running..." : "N/A";
+  }
+
+  getDuration(job: any): string {
+    // Find the timestamp with status 3 (success), 4 (error), or 5 (cancelled)
+    const endTimestamp = job.timestamps?.find((ts: any) =>
+      [3, 4, 5].includes(ts.status),
+    );
+
+    if (endTimestamp) {
+      const startDate = new Date(job.created).getTime();
+      const endDate = new Date(endTimestamp.time).getTime();
+      const duration = endDate - startDate;
+      return formatDuration(duration);
+    }
+
+    if (job.status === 2) {
+      // For running jobs, calculate duration from start until now
+      const startDate = new Date(job.created).getTime();
+      const now = new Date().getTime();
+      const duration = now - startDate;
+      return formatDuration(duration) + " (running)";
+    }
+
+    return "N/A";
+  }
+
+  async viewJobLog(job: IJob): Promise<void> {
+    this.selectedJob = job;
+
+    // Show loading in the log area if not refreshing
+    if (!this.refreshingLog) {
+      this.currentJobLog = "Loading job log...";
+      this.showLogDialog = true;
+    }
+
+    try {
+      // Fetch the job info with log
+      const jobWithLog = await main.api.getJobInfo(job._id);
+
+      if (!jobWithLog) {
+        this.currentJobLog = "Failed to load job log.";
+        return;
+      }
+
+      // Get end time and duration for the log header
+      const endTime = this.getEndTime(jobWithLog);
+      const duration = this.getDuration(jobWithLog);
+
+      // Create a formatted log header
+      const logHeader =
+        `=== Job ${jobWithLog._id} (${jobWithLog.title}) ===\n\n` +
+        `Started: ${formatDateString(jobWithLog.created)}\n` +
+        `Ended: ${endTime}\n` +
+        `Duration: ${duration}\n` +
+        `Status: ${this.getStatusText(jobWithLog.status)}\n` +
+        `Type: ${jobWithLog.type}\n\n` +
+        `Arguments:\n${jobWithLog.args.join("\n")}\n\n` +
+        `${
+          jobWithLog.status === 3
+            ? "Job completed successfully."
+            : jobWithLog.status === 4
+              ? "Job failed with errors."
+              : jobWithLog.status === 2
+                ? "Job is still running..."
+                : "Job status: " + this.getStatusText(jobWithLog.status)
```

Comment:
Seeing that, you probably should add a property to your Interface:
```
interface JobLogProperty {
  color: string;
  statusText: string;
  stateText ?: string;
}
```
then, update your object:
```
const jobLogStatus: {[key: number]: JobLogProperty} = {
  0: {color: "grey", statusText:  "Inactive"},
  1: {color: "blue", statusText:  "Queued"},
  2: {color: "orange", statusText:  "Running", stateText: "Job is still running..."},
  ....
}
```
and define a function:
```
function getJobState(status: number): string {
  if (status in jobLogStatus && jobLogStatus[status].stateText) {
    return  jobLogStatus[status].stateText
  }
  return "Job status: " + getStatusText(status)
}
```

and finally here, you should call `getJobState(jobWithLog.status)`

Reply:
Thanks, done!